Sources/Core/AWSSDKIdentity/Sources/AWSSDKIdentity/IdentityProvidingClients/InternalClientKeys.swift

```diff
@@ -0,0 +1,15 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import struct Smithy.AttributeKey
+
+public enum InternalClientKeys {
+    /// The STS client to be used in credential resolution.
+    public static let internalSTSClientKey = AttributeKey<any IdentityProvidingSTSClient>(
+        name: "IdentityProvidingSTSClient"
+    )
+}
```

Comment:
This attribute key is used by auth scheme resolvers (generated for each public service module) to save an instance of `IdentityProvidingSTSClient` into `identityProperties` attributes of auth option.